File: Parenthesis Checker.cpp
class Solution
{
    public:
    
    bool ispar(string x)
    {
        // Your code here
        stack<char>st;
        
        for(auto ch : x){
            
            if(st.empty() && (ch==')' || ch =='}' || ch==']')){
                return false;
            }
            if(ch=='(' || ch=='{' || ch=='['){
                st.push(ch);
            }else{
                char top = st.top();
                if(top=='[' && ch==']'){
                    st.pop();
                }
                else if(top=='{' && ch=='}'){
                    st.pop();
                }
                else if(top=='(' && ch==')'){
                    st.pop();
                }else{
                    return false;
                }
            }
        }
        if(!st.empty()){
            return false;
        }
        return true;
    }

};

File: Parenthesis Checker.py
class Solution:

    def isBalanced(self, s):

        # code here

        n = len(s)

        stack = []

        for i in range(n):

            curr = s[i]

            if curr == '(' or curr == '{' or curr == '[':

                stack.append(curr)

            else:

                if not stack:

                    return False

                else:

                    top = stack[-1]

                    stack.pop()

                    if (curr == ')' and top != '(') or (curr == '}' and top != '{') or (curr == ']' and top != '['):

                        return False

        return True if len(stack) == 0 else False
